be/src/fs/s3/poco_common.cpp

```diff
@@ -97,4 +81,98 @@ bool checkRequestCanReturn2xxAndErrorInBody(Aws::Http::HttpRequest& request) {
     return false;
 }
 
+HTTPSessionPtr makeHTTPSessionImpl(const std::string& host, Poco::UInt16 port, bool https, bool keep_alive) {
+    HTTPSessionPtr session;
+
+    if (https) {
+        session = std::make_shared<Poco::Net::HTTPSClientSession>(host, port);
+    } else {
+        session = std::make_shared<Poco::Net::HTTPClientSession>(host, port);
+    }
+
+    // doesn't work properly without patch
+    session->setKeepAlive(keep_alive);
+    return session;
+}
+
+class EndpointHTTPSessionPool : public PoolBase<Poco::Net::HTTPClientSession> {
+public:
+    using Base = PoolBase<Poco::Net::HTTPClientSession>;
+    EndpointHTTPSessionPool(const std::string& host, uint16_t port, bool is_https)
+            : Base(ENDPOINT_POOL_SIZE), _host(host), _port(port), _is_https(is_https) {}
+
+private:
+    ObjectPtr allocObject() override {
+        SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(GlobalEnv::GetInstance()->poco_connection_pool_mem_tracker());
```

Comment:
provide interface to set, instead of get. to avoid cyclic reference.

Reply:
will do it.